File: JWP_API/JadeWesserPort/DTOs/PlanningDTOs/RebalanceComparisonDto.cs
using JadeWesserPort.DTOs.DailyScheduleDTOs;

namespace JadeWesserPort.DTOs.PlanningDTOs;

public class RebalanceComparisonDto
{
    public DailyScheduleResponseDto OldSchedule { get; set; } = new();
    public DailyScheduleResponseDto NewSchedule { get; set; } = new();
    public List<VesselDockChangeDto> DockChanges { get; set; } = []; 
}

File: Frontend/src/infrastructure/dtos/reabalance-comparison/rebalanceComparisonDto.ts
import { DailyScheduleResponseDto } from "../scheduling/DailyScheduleResponseDto";
import type { VesselDockChangeDto } from "./vesselDockChangeDto";

export type RebalanceComparisonDto = {
  oldSchedule: DailyScheduleResponseDto;
  newSchedule: DailyScheduleResponseDto;
  dockChanges: VesselDockChangeDto[];
};